src/pages/details/index.jsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+import { getProductsById } from '../../redux/detailsSlice';
+import {
+  Container,
+  ImageContainer,
+  Image,
+  DetailsContainer,
+  Title,
+  Description,
+  Price,
+  Category,
+  Rate,
+} from './styles/details';
+
+const Details = () => {
+  const dispatch = useDispatch();
+  const { pathname } = useLocation();
+  const detailsReducer = useSelector((state) => state.detailsReducer);
+  const { singleProduct: product, isActive: isOpen } = detailsReducer;
+  const { category, description, image, price, rating, title } = product;
+  const productId = pathname.split('/')[3];
+
+  useEffect(() => {
+    dispatch(getProductsById(productId));
+  }, [dispatch, productId]);
+
+  return (
+    <Container isActive={isOpen}>
+      <ImageContainer>
+        <Image src={image} alt={title} />
+      </ImageContainer>
+      <DetailsContainer>
+        <Title>{title}</Title>
+        <Description>{description}</Description>
+        <Price>Price: ${price}</Price>
+        <Rate>Rate: {rating?.rate}</Rate>
+        <Category>Category: {category}</Category>
+      </DetailsContainer>
+    </Container>
```

Comment:
- [ ] Kindly follow the given design, this includes font, typography and layout, as this is a requirement
![image](https://user-images.githubusercontent.com/81932478/211147905-a8080692-a53a-4446-8055-e07c88843a2d.png)

Your design
![image](https://user-images.githubusercontent.com/81932478/211147359-b93eec72-cf32-4013-b798-09b35a18aa7d.png)

Given design
![image](https://user-images.githubusercontent.com/81932478/211147426-806f3e9b-b2c5-4ae8-95a9-8a08b615482c.png)